src/lightning_app/testing/testing.py

```diff
@@ -253,29 +259,44 @@ def run_app_in_cloud(
             env_copy["LIGHTNING_DEBUG"] = "1"
         shutil.copytree(app_folder, tmpdir, dirs_exist_ok=True)
         # TODO - add -no-cache to the command line.
-        process = Popen(
-            (
-                [
-                    sys.executable,
-                    "-m",
-                    "lightning",
-                    "run",
-                    "app",
-                    app_name,
-                    "--cloud",
-                    "--name",
-                    name,
-                    "--open-ui",
-                    "false",
-                ]
-                + extra_args
-            ),
-            cwd=tmpdir,
-            env=env_copy,
-            stdout=sys.stdout,
-            stderr=sys.stderr,
-        )
-        process.wait()
+        stdout_path = get_logfile("running_process")
+        with open(stdout_path, "w") as stdout:
+            process = Popen(
+                (
+                    [
+                        sys.executable,
+                        "-m",
+                        "lightning",
+                        "run",
+                        "app",
+                        app_name,
+                        "--cloud",
+                        "--name",
+                        name,
+                        "--open-ui",
+                        "false",
+                    ]
+                    + extra_args
+                ),
+                cwd=tmpdir,
+                env=env_copy,
+                stdout=stdout,
+                stderr=sys.stderr,
+            )
+            process.wait()
+
+        if is_editable_mode:
+            # Added to ensure the current code is properly uploaded.
+            # Otherwise, it could result in un-tested PRs.
+            has_found = False
+            with open(stdout_path) as f:
+                for line in f.readlines():
+                    if "Packaged Lightning with your application" in line:
+                        has_found = True
+                    print(line)
+
+            assert has_found
+        os.remove(stdout_path)
```

Comment:
so this would work only if you install it with `-e` which is not the default option, shall we as else here raise a warning that we could not verify that the package was uploaded correctly?

Reply:
For Lightning App, the package needs to be installed with -e